### PR TITLE
Added Require GPA As Event Option 

### PR DESCRIPTION
**Pull request by subhankar-panda**
This PR also adds functionality to show the GPA fields as updatable on Admin and User profiles.

### CODE REVIEW

File: src/assets/js/pages/ApplyPage/index.js [hunk 1]
```diff
@@ -229,7 +229,7 @@ class ApplyPage extends React.Component {
       <div className="page apply-page">
         <NavHeader />
         <div className="sd-form__wrapper">
-          <Progress percent={(page * 100) / 4} />
+          <Progress percent={(page * 100) / 4} style={{zIndex: 100}} />
```
Comment (RedbackThomson) [inline src/assets/js/pages/ApplyPage/index.js]: Should we give this its own class instead?

File: src/assets/js/pages/ApplyPage/validate.js [hunk 2]
```diff
@@ -28,6 +28,10 @@ const createValidator = (options) => (values) => {
     required.push('race');
   }
 
+  if (options.requireGPA) {
+    required.push('gpa')
```
Comment (RedbackThomson) [inline src/assets/js/pages/ApplyPage/validate.js]: Does not require the Major GPA? Perhaps they should be 2 separate options.

File: src/server/models/user.js [hunk 3]
```diff
@@ -140,9 +140,20 @@ var UserSchema = new Schema({
...
   race: {
     type: String,
     required: false
+  },
+  // Declares the user's GPA, required by events on an optional basis
+  gpa: {
+    type: String,
+    required: false
+  },
+  // Declares the user's GPA, required by events on an optional basis
```
Comment (RedbackThomson) [inline src/server/models/user.js]: This comment is the same as above.

**Comment by subhankar-panda** [comment]
Merged into #102

**Pull request closed without merging**